kafka-topic-creator/helm/values.yaml

```diff
@@ -41,6 +41,8 @@ ttlSecondsAfterFinished: 900
 
 kafka:
   address: "bootstrap:9092"
+  # for each integer value based key provided in topic config, override value to max(in override, provided value)
+  minValueOverrideForTopicConfig: {}
```

Comment:
@kishansairam9 not clear what will this change do? can you give an example?

Reply:
Sure. Consider this sample topics config
```yaml
  topics:
    alpha:
      replicationFactor: 3
      partitions: 8
      configs:
        max.message.bytes: 4000
        retention.bytes: 9000
        retention.ms: 1000
    beta:
      replicationFactor: 3
      partitions: 8
      configs:
        max.message.bytes: 1000
        retention.bytes: 5000
        retention.ms: 1000
```
Now for example we want to have a minimum retention of at least 7000 bytes for all topics, we need to go to each topic in values.yaml and update where it is lower. It is tedious, also error prone in case of MiB values instead of 0s.

With this new support of minimum value overrides per config key, we can do that in one shot
```yaml
minValueOverrideForTopicConfig:
    retention.bytes: 7000
    retention.ms: 3000
```

```diff
52c52 // alpha
<               "max.message.bytes=4000","retention.bytes=9000","retention.ms=1000" \
---
>               "max.message.bytes=4000","retention.bytes=9000","retention.ms=3000" \
67c67 // beta
<               "max.message.bytes=1000","retention.bytes=5000","retention.ms=1000" \
---
>               "max.message.bytes=1000","retention.bytes=7000","retention.ms=3000" \
```

Reply:
I changing stuff at broker config level might have similar effect. But we can bake in this functionality via helm into topic creator instead of touch broker configs, which will apply these overrides only for those managed by topic creator.